### ISSUE TITLE
Inference of class variable types with --verifytypes

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#1493]
**Describe the bug**
pyright isn't inferring types of class variables when doing `verifytypes`

**To Reproduce**
Create a directory called `callib` with empty `__init__.py` and `py.typed` files.  Create `fooclass.py` with the following:

```python
from typing import Any, Optional, Callable


def make_invalid_op(name: str) -> Callable[[Any, Optional[Any]], Any]:
    """A docstring"""

    def invalid_op(self, other=None):
        raise TypeError(f"invalid for {name}")

    return invalid_op


def stringfunc() -> str:
    """A docstring"""
    return "hello"


class Foo:
    """A test class"""

    def __init__(self):
        pass

    __add__ = make_invalid_op("add")

    a_str = stringfunc()
```

Use `python --verifytypes callib` and get the following:

```
Package name: "callib"
Package directory: "callib"
Path of py.typed file: "callib\py.typed"
  error: Type partially unknown for class "callib.fooclass.Foo"
    Type partially unknown for class "Foo"
      Type unknown for variable "__add__"
      Type unknown for variable "a_str"
  error: Type not declared for variable "callib.fooclass.Foo.__add__"
  error: Type not declared for variable "callib.fooclass.Foo.a_str"
```

**Expected behavior**
Types are inferred for the class variables.

**VS Code extension or command-line**
Command line, pyright verssion 1.1.110

**Additional context**
This came up first with the `Callable` example when analyzing `pandas` code, but then I tried the simple example of a function returning a string.

**Comment by erictraut** [comment]
This is correct behavior. Type inference is not used when a library is "py.typed" because inference rules are not standardized across type checkers, and it's important for consumers of a library to get the same behavior regardless of which type checker they're using. All public class variables need to have explicit type annotations. If the class variable is not intended to be part of the public interface, then it should be renamed so it starts with an underscore character.

**Comment by Dr-Irv** [comment]
> All public class variables in a "py.typed" library need to have explicit type annotations.

Is this specified in a PEP somewhere?  Or something you've decided that `pyright` will do?

**Comment by erictraut** [comment]
It falls out logically from the goals and requirements. The goal of type annotations within a library is to enable consistent and accurate type checking for consumers of that library. The behavior shouldn't vary between type checkers or other tools that consume this type information.

The goals and the rules are documented [here](https://github.com/microsoft/pyright/blob/master/docs/typed-libraries.md). This hasn't been formalized in a PEP, but it has been discussed at length in the typing-sig, and there is general agreement among the type checking vendors.
